### PR TITLE
Skip wcsaxes tests if pytest-mpl is not installed

### PR DESCRIPTION
**Pull request by larrybradley**
After installing with `pip install -e .[test]`, running `pytest` manually in the `astropy/visualization` gives a bunch of `pytest.PytestUnknownMarkWarning: Unknown pytest.mark.mpl_image_compare` errors, preventing the tests from running.  This subpackage requires the `pytest-mpl` to be installed.  I added that package to the setup in the `test` section.

### CODE REVIEW

**Comment by larrybradley** [comment]
Yeah, @pllim mentioned that's why `pytest-mpl` wasn't included in `pytest-astropy`.

I just pushed a better fix where the `wcsaxes` tests are skipped if `pytest-mpl` is not installed.

File: astropy/visualization/wcsaxes/__init__.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 try:
     import pytest
     pytest.importorskip("matplotlib")
+    pytest.importorskip("pytest-mpl")
```
Comment (bsipocz) [inline astropy/visualization/wcsaxes/__init__.py]: Shouldn't these live in the test files rather than in the module itself? I'm fine with the fix as it's just extends the current approach, but wonder whether the current approach is optimal or not.

Comment (larrybradley) [reply astropy/visualization/wcsaxes/__init__.py]: I was confused too....wondering how the `wcsaxes` tests are skipped if `mpl` isn't installed because they are written such that they *require* `mpl`.  They do not check if `mpl` is available, they just import `mpl`.  I discovered how the tests are skipped in this `__init__` file.  We could replace this by adding the `mpl` checks, but the entire `visualization.wcsaxes` subpackage *requires* `mpl`  @astrofrog wrote this, so I'll let him comment.

**Comment by astrofrog** [comment]
I'll review this shortly!

File: docs/visualization/matplotlib_integration.rst [hunk 2]
```diff
@@ -69,7 +69,7 @@ precise scale and format to use for the tick labels, in which case you can make
 use of the `~astropy.visualization.time_support` function. This feature needs to
 be explicitly turned on:
 
-.. doctest-requires:: matplotlib
+.. doctest-requires:: matplotlib, pytest-mpl
```
Comment (astrofrog) [inline docs/visualization/matplotlib_integration.rst:72]: I'm confused - why are these needed? the doctests don't use pytest-mpl?

Comment (larrybradley) [reply docs/visualization/matplotlib_integration.rst:72]: The doctests apparently do use `pytest-mpl`.  Here's the traceback I get without requiring `pytest-mpl`:

```
――――――――――― [doctest] docs/visualization/matplotlib_integration.rst ――――――――――――
066
067 However, in many cases, you will probably want to have more control over the
068 precise scale and format to use for the tick labels, in which case you can make
069 use of the `~astropy.visualization.time_support` function. This feature needs to
070 be explicitly turned on:
071
072 .. doctest-requires:: matplotlib
073
074     >>> from astropy.visualization import time_support
075     >>> time_support()  # doctest: +IGNORE_OUTPUT
UNEXPECTED EXCEPTION: could not import 'pytest-mpl': No module named 'pytest-mpl'
Traceback (most recent call last):

  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py", line 1330, in __run
    compileflags, 1), test.globs)

  File "<doctest matplotlib_integration.rst[3]>", line 1, in <module>

  File "/Users/lbradley/Dropbox/softw/packages/dev/astropy/astropy/visualization/time.py", line 50, in time_support
    from astropy.visualization.wcsaxes.utils import select_step_hour, select_step_scalar

  File "/Users/lbradley/Dropbox/softw/packages/dev/astropy/astropy/visualization/wcsaxes/__init__.py", line 9, in <module>
    pytest.importorskip("pytest-mpl")

  File "/Users/lbradley/.venvs/dev/lib/python3.7/site-packages/_pytest/outcomes.py", line 193, in importorskip
    raise Skipped(reason, allow_module_level=True) from None

Skipped: could not import 'pytest-mpl': No module named 'pytest-mpl'

/Users/lbradley/Dropbox/softw/packages/dev/astropy/docs/visualization/matplotlib_integration.rst:75: UnexpectedException
```

**Comment by larrybradley** [comment]
@astrofrog I'm not sure how to register a mock marker in conftest.  If you have a better solution, please go ahead and implement it and close this PR.

File: docs/visualization/matplotlib_integration.rst (hunk 2 above)
Comment (bsipocz) [reply docs/visualization/matplotlib_integration.rst:72]: yes, I don't think any pytest.importskips should be in the `__init__.py` files. And as pytest is not a mandatory runtime dependency any more, I'm having some issues with the matplotlib importskip approach, too.

**Comment by astrofrog** [comment]
I can try and take a look next week.

**Comment by astrofrog** [comment]
The bot had an issue closing this. I'll close it manually and see if I can open an alternative PR

**Comment by astrofrog** [comment]
Actually the warnings no longer occur as we filter that warning in setup.cfg - setting up a dummy mark is not worth it - I just tried and it adds complexity to ``conftest.py``.
